### ISSUE TITLE
Regression: Table metadata calculation for Oracle does not work after 3.19.0

### ISSUE THREAD

**Issue by vitaly-ivanov** [jOOQ/jOOQ#16054]
### Expected behavior

Metadata is calculated correctly.

### Actual behavior

The incorrect query is running while calculating metadata. Stacktrace:

```
org.jooq.exception.DataAccessException: SQL [select null catalog_name, t.OWNER, t.TRIGGER_NAME, null table_catalog_name, t.TABLE_OWNER, t.TABLE_NAME, case when t.TRIGGERING_EVENT like '%INSERT%' then true when not (t.TRIGGERING_EVENT like '%INSERT%') then false end i, case when t.TRIGGERING_EVENT like '%UPDATE%' then true when not (t.TRIGGERING_EVENT like '%UPDATE%') then false end u, case when t.TRIGGERING_EVENT like '%DELETE%' then true when not (t.TRIGGERING_EVENT like '%DELETE%') then false end d, case when t.TRIGGER_TYPE like '%EACH ROW%' then 'FOR_EACH_ROW' when t.TRIGGER_TYPE like '%STATEMENT%' then 'FOR_EACH_STATEMENT' end action_orientation, case when t.TRIGGER_TYPE like '%BEFORE%' then 'BEFORE' when t.TRIGGER_TYPE like '%AFTER%' then 'AFTER' when t.TRIGGER_TYPE like '%INSTEAD OF%' then 'INSTEAD_OF' end action_timing, t.WHEN_CLAUSE, 0 action_order, t.TRIGGER_BODY from SYS.ALL_TRIGGERS t where (t.OWNER in (?) and t.BASE_OBJECT_TYPE in ('TABLE', 'VIEW')) order by t.OWNER, t.TRIGGER_NAME]; ORA-00904: "FALSE": invalid identifier

https://docs.oracle.com/error-help/db/ora-00904/
	at org.jooq_3.19.0.ORACLE.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3527)
	at org.jooq.impl.Tools.translate(Tools.java:3515)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:812)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:290)
	at org.jooq.impl.MetaImpl$MetaSchema.lambda$init0$8(MetaImpl.java:569)
	at org.jooq.impl.MetaImpl.withCatalog(MetaImpl.java:236)
	at org.jooq.impl.MetaImpl$MetaSchema.lambda$init0$9(MetaImpl.java:561)
	at org.jooq.impl.MetaImpl.lambda$meta$0(MetaImpl.java:207)
	at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:657)
	at org.jooq.impl.MetaImpl.meta(MetaImpl.java:207)
	at org.jooq.impl.MetaImpl$MetaSchema.init0(MetaImpl.java:560)
	at org.jooq.impl.MetaImpl$MetaSchema.initTriggers(MetaImpl.java:654)
	at org.jooq.impl.MetaImpl$MetaSchema.getTriggers(MetaImpl.java:645)
	at org.jooq.impl.MetaImpl$MetaSchema.lambda$getTables$2(MetaImpl.java:508)
	at org.jooq.impl.Tools.map(Tools.java:2508)
	at org.jooq.impl.MetaImpl$MetaSchema.getTables(MetaImpl.java:477)
	at org.jooq.impl.FilteredMeta$FilteredSchema.getTables(FilteredMeta.java:395)
```

### Steps to reproduce the problem

- Create table in the "TEST" schema:
```sql
CREATE TABLE table3
(
    pk0    integer  NOT NULL,
    amount BINARY_DOUBLE  NOT NULL
);
```

- Exception occurs while calling the following code:
```kotlin
dslContext.meta().schemas
        .find { it.name == "TEST" }
        ?.tables
        ?.find { it.name == table.name }
```

### jOOQ Version

jOOQ Professional  3.19.0

### Database product and version

Oracle image gvenzl/oracle-xe:21.3.0-slim

### Java Version

17

### OS Version

Any

### JDBC driver name and version (include name if unofficial driver)

com.oracle.database.jdbc:ojdbc11:23.3.0.23.09

**Comment by lukaseder** [comment]
Thanks a lot for your report.

The unversioned dialect family `SQLDialect.ORACLE` always corresponds to the latest supported Oracle version, in case of jOOQ 3.19, that's `SQLDialect.ORACLE23C`, which introduce a lot of new features. Since you're not using 23c yet, I suggest you use a more appropriate `SQLDialect` version, e.g. `SQLDialect.ORACLE21C`.

However, the problem still persists in that case, as the SQL query doesn't seem to change based on dialect version, in this case. I will look into this right away.

As a workaround, you could patch the SQL contained in the `org.jooq.impl.MetaSQL` class, e.g. using reflection.

**Comment by lukaseder** [comment]
`Meta.getSequences()` is also affected

**Comment by lukaseder** [comment]
Related issue:

- https://github.com/jOOQ/jOOQ/issues/16059

**Comment by lukaseder** [comment]
Fixed in versions

- 3.20.0
- 3.19.2 (https://github.com/jOOQ/jOOQ/issues/16063)

**Comment by vitaly-ivanov** [comment]
@lukaseder Thank you for the quick fix, it works on 3.19.2-SNAPSHOT.

**Comment by lukaseder** [comment]
Thanks a lot for the confirmation. I'll release 3.19.2 today.
